File: sdk/csharp/src/Exceptions/AuthenticationException.cs
using System.Net;

namespace SuperDev.SDK.Exceptions;

/// <summary>
/// Raised when authentication or authorization fails (HTTP 401 or 403).
/// </summary>
public class AuthenticationException : SuperDevException
{
    public AuthenticationException(string message = "Authentication failed", Dictionary<string, object>? details = null)
        : base(message, HttpStatusCode.Unauthorized, details)
    {
    }
}

File: sdk/java/src/main/java/com/superdev/sdk/exceptions/AuthenticationException.java
package com.superdev.sdk.exceptions;

/**
 * Thrown when authentication fails (e.g., invalid or missing API key).
 */
public class AuthenticationException extends SuperDevException {

    /**
     * Creates a new AuthenticationException.
     *
     * @param message the error message
     */
    public AuthenticationException(String message) {
        super(message, 401);
    }

    /**
     * Creates a new AuthenticationException with a cause.
     *
     * @param message the error message
     * @param cause   the underlying cause
     */
    public AuthenticationException(String message, Throwable cause) {
        super(message, cause);
    }
}
